conjure-lib/src/main/java/com/palantir/conjure/java/lib/Bytes.java

```diff
@@ -128,4 +139,56 @@ public Bytes deserialize(JsonParser parser, DeserializationContext _ctxt) throws
             return new Bytes(parser.getBinaryValue());
         }
     }
+
+    public static final class Builder extends OutputStream {
+        private byte[] contents;
+        private boolean isBuilt;
+        private int count;
+
+        public Builder() {
+            this(32);
+        }
+
+        public Builder(int capacity) {
+            Preconditions.checkArgument(capacity > 0);
+            this.contents = new byte[capacity];
+            this.count = 0;
+        }
+
+        @Override
+        public synchronized void write(int inputByte) {
+            Preconditions.checkState(!isBuilt, "No writes are allowed because Immutable Bytes have already been built");
+            ensureCapacity(count + 1);
+            contents[count] = (byte) inputByte;
+            count += 1;
+        }
+
+        @Override
+        public synchronized void write(byte[] bytes, int off, int len) {
+            Preconditions.checkState(!isBuilt, "No writes are allowed because Immutable Bytes have already been built");
+            Objects.checkFromIndexSize(off, len, bytes.length);
+            ensureCapacity(count + len);
+            System.arraycopy(bytes, off, contents, count, len);
+            count += len;
+        }
+
+        public Bytes build() {
+            close();
+            return new Bytes(contents);
```

Comment:
This needs to be resized to remove uninitialized values.

Unless you know the exact size of the byte array ahead of time, I don't think we can avoid a copy here.

Reply:
A more broadly applicable approach may be to define a `from(ByteArrayOutputStream)`. `ByteArrayOutputStream::toByteArray` makes a copy via `Arrays::copyOf` so this returns an immutable copy.

Reply:
This wouldn't allow you do to anything that you couldn't do with the existing API.

Reply:
Sure, I can extend `ByteArrayOutputStream` to expose the `protected` fields `buf[]` and `count` and use `Bytes::from(byte[] array, int offset, int length)`. I'll do that for now.